modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -273,9 +275,25 @@ package object widgets {
     encoded
   } else Nil
 
+  def fromDFToPoint(df:DataFrame, maxPoints:Int):Seq[MagicRenderPoint] = {
+    val rows = df.take(maxPoints)
+    if (rows.nonEmpty) {
+      val points = df.schema.toList.head match {
```

Comment:
I am referring to the string test; if the first column is a string, all columns must be strings?

Sent from my phone.
________________________________
From: Andy Petrella<mailto:notifications@github.com>
Sent: ‎6/‎14/‎2015 12:43 PM
To: andypetrella/spark-notebook<mailto:spark-notebook@noreply.github.com>
Cc: Eron Wright<mailto:ewright@live.com>
Subject: Re: [spark-notebook] First iteration of simple DataFrame auto (or not) rendering (fixes #200) (#214)

> @@ -273,9 +275,25 @@ package object widgets {
>      encoded
>    } else Nil
>
> +  def fromDFToPoint(df:DataFrame, maxPoints:Int):Seq[MagicRenderPoint] = {
> +    val rows = df.take(maxPoints)
> +    if (rows.nonEmpty) {
> +      val points = df.schema.toList.head match {

the case with rows is necessary in case the df is empty, which could happen.

`df.toJSON` is interesting to use, the sad thing is that it returns a `RDD[String]` so it''s lazy and we loose metadata (schema). It was hard to squeeze it literally and keeping the code simple (again, for a first cut)

---
Reply to this email directly or view it on GitHub:
https://github.com/andypetrella/spark-notebook/pull/214/files#r32383458